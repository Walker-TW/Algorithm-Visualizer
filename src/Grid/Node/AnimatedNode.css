.Node {
  width: 20px;
  height: 20px;
  border: 1px solid black;
  background-color: grey;
  transition: transform 0.2s;
}

.visited {
  animation-name: visitedAnimation;
  animation-duration: 0.7s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes visitedAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background: radial-gradient(circle, #739da5, #020024, #090979);
  }
}

.start {
  background-color: rgb(206, 47, 255);
  animation-name: startAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes startAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: rgb(206, 47, 255);
  }
}

.finish {
  background-color: yellow;
  animation-name: finishAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes finishAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: yellow;
  }
}

.path {
  background-color: chartreuse;
  animation-name: pathAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes pathAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(186, 24, 178);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(247, 191, 70);
  }

  100% {
    transform: scale(1);
    background-color: chartreuse;
  }
}

.fence {
  animation-name: fenceAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}

@keyframes fenceAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: rgb(0, 0, 0);
  }
}
