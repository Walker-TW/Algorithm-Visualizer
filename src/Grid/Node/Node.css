.Node {
  width: 20px;
  height: 20px;
  border: 1px solid black;
  background-color: lightcyan;
}

.visited {
  background-color: hotpink;
}

.start {
  background-color: rgb(206, 47, 255);
}

.finish {
  background-color: yellow;
}

.path {
  background-color: chartreuse;
}
