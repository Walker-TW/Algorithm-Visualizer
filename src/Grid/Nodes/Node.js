import React from 'react';

const Node = () => {
  const getStyle = {
    width: '20px',
    height: '20px',
    border: '1px solid black',
  };

  return <div style={getStyle} />;
};

export default Node;
