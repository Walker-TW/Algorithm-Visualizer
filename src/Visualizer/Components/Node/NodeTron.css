.Node {
  width: 20px;
  height: 20px;
  border: 2px solid rgb(189, 189, 5);
  /* background-color: black; */
  /* opacity: 0.5; */
  transition: transform 0.2s;
  border-radius: 2px;
}

@media (max-width: 1030px) {
  .Node {
    width: 30px;
    height: 30px;
  }
}

@media (max-width: 500px) {
  .Node {
    width: 40px;
    height: 40px;
  }
}

.passage {
  background-color: olive;
}
.visited {
  animation-name: visitedAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes visitedAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(255, 4, 100);
  }

  50% {
    transform: scale(1.3);
    background-color: rgb(218, 157, 27);
  }

  100% {
    transform: scale(1);
    background-color: rgb(255, 225, 0);
    border: 1px solid black;
  }
}

.start {
  background-color: rgb(206, 47, 255);
  animation-name: startAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes startAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: rgb(206, 47, 255);
  }
}

.finish {
  background-color: yellow;
  animation-name: finishAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes finishAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: yellow;
  }
}

.path {
  animation-name: pathAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes pathAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(186, 24, 178);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(247, 191, 70);
  }

  100% {
    transform: scale(1);
    background-color: rgb(0, 174, 255);
  }
}

.fence {
  /* background-color: orange; */
  animation-name: fenceAnimation;
  animation-duration: 0.1s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}

@keyframes fenceAnimation {
  /* 0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  } */

  0% {
    transform: scale(1.3);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: rgb(255, 0, 0);
  }
}
