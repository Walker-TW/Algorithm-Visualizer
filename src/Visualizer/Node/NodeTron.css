.Node {
  width: 20px;
  height: 20px;
  border: 2px solid rgb(189, 189, 5);
  /* background-color: black; */
  /* opacity: 0.5; */
  transition: transform 0.2s;
  border-radius: 2px;
}

@media (max-width: 900px) {
  .Node {
    width: 30px;
    height: 30px;
  }
}

@media (max-width: 500px) {
  .Node {
    width: 40px;
    height: 40px;
  }
}

.visited {
  animation-name: visitedAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes visitedAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(19, 169, 129);
  }

  /* 15% {
    transform: scale(0.7);
    background-color: rgb(22, 186, 63);
  } */
  /* 25% {
    transform: scale(1);
    background-color: rgb(51, 225, 25);
  } */

  50% {
    transform: scale(1.3);
    background-color: rgb(218, 157, 27);
  }

  /* 75% {
    transform: scale(1.1);
    background-color: rgb(189, 114, 0);
  } */

  /* 80% {
    transform: scale(1.2);
    background-color: rgb(145, 128, 2);
  } */

  100% {
    transform: scale(1);
    /* background: radial-gradient(circle, yellow, #9b9ba3, #090979); */
    background-color: rgb(255, 225, 0);
    border: 1px solid black;
  }
}

.start {
  background-color: rgb(206, 47, 255);
  animation-name: startAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes startAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: rgb(206, 47, 255);
  }
}

.finish {
  background-color: yellow;
  animation-name: finishAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes finishAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: yellow;
  }
}

.path {
  background-color: chartreuse;
  animation-name: pathAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}
@keyframes pathAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(186, 24, 178);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(247, 191, 70);
  }

  100% {
    transform: scale(1);
    background-color: chartreuse;
  }
}

.fence {
  animation-name: fenceAnimation;
  animation-duration: 0.3s;
  animation-timing-function: ease-out;
  animation-delay: 0;
  animation-direction: alternate;
  animation-iteration-count: 1;
  animation-fill-mode: forwards;
  animation-play-state: running;
}

@keyframes fenceAnimation {
  0% {
    transform: scale(0.3);
    background-color: rgb(23, 154, 45);
  }

  50% {
    transform: scale(1.2);
    background-color: rgb(194, 142, 28);
  }

  100% {
    transform: scale(1);
    background-color: rgb(255, 0, 0);
  }
}
