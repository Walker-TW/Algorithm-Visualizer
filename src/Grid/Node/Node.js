import React, { Component } from "react";
import "./Node.css";
import PropTypes from "prop-types";

export default class Node extends Component {
  state = {
    start: false,
    finish: false,
    fence: false,
  };

  clickHandler = () => {
    const {
      gridId,
      gridHasStart,
      gridHasFinish,
      gridHasFenceToggle,
      nodeFlag,
    } = this.props;

    if (gridHasStart && gridHasFinish && !gridHasFenceToggle) {
      document.getElementById("button").className = "btn btn-primary big";
    } else {
      if (!gridHasStart) {
        nodeFlag(gridId, "start");
        this.setState({ start: true });
      } else {
        this.setState({ finish: true });
        nodeFlag(gridId, "finish");
      }
    }
  };

  fenceSelector = (e) => {
    const { gridId, nodeFlag } = this.props;
    const { fence } = this.state;
    if (fence === false) {
      nodeFlag(gridId, "fence");
      this.setState({ fence: true });
    } else {
      nodeFlag(gridId, "fence");
      this.setState({ fence: false });
    }
  };

  mouseUpHandler = () => {
    const { mouseFlag, mouseToggle, gridHasFenceToggle } = this.props;
    if (gridHasFenceToggle === true && mouseToggle === true) {
      mouseFlag();
    }
  };
  mouseEnterHandler = () => {
    const { mouseToggle, gridHasFenceToggle } = this.props;
    if (gridHasFenceToggle === true && mouseToggle === true) {
      this.fenceSelector();
    }
  };
  mouseDownHandler = () => {
    const { mouseToggle, mouseFlag, gridHasFenceToggle } = this.props;
    if (gridHasFenceToggle === true && mouseToggle === false) {
      mouseFlag();
      this.fenceSelector();
    } else {
      this.clickHandler();
    }
  };

  render() {
    return (
      <div
        onMouseDown={this.mouseDownHandler}
        onMouseEnter={this.mouseEnterHandler}
        onMouseUp={this.mouseUpHandler}
        className={`Node ${
          this.state.start
            ? "start"
            : this.state.finish
            ? "finish"
            : this.state.fence
            ? "fence"
            : ""
        }`}
        id={this.props.id}
      />
    );
  }
}

Node.propTypes = {
  gridId: PropTypes.object.isRequired,
  nodeFlag: PropTypes.func.isRequired,
  gridHasStart: PropTypes.bool.isRequired,
  gridHasFinish: PropTypes.bool.isRequired,
  gridHasFenceToggle: PropTypes.bool.isRequired,

  // reset: PropTypes.func.isRequired,
};
