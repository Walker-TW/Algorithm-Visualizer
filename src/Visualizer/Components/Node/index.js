import React, { Component } from 'react';
import PropTypes from 'prop-types';

// import "./Node.css";
import './NodeTron.css';
// import './AnimatedNode.css';
class Node extends Component {
  state = {
    start: false,
    finish: false,
    fence: false,
  };

  // start and end points are not adjustable
  // at the moment so they have a seperate handler

  startFinishHandler = (type) => {
    const { gridId, nodeFlag } = this.props;
    nodeFlag(gridId, [type]);
    this.setState({ [type]: true });
  };

  fenceHandler = () => {
    const { gridId, nodeFlag } = this.props;
    const { fence, start, finish } = this.state;

    if (!start && !finish) nodeFlag(gridId, 'fence');
    this.setState({ fence: !fence });
  };

  mouseUpHandler = () => {
    const { mouseFlag, mouseToggle, fenceToggle } = this.props;

    if (fenceToggle && mouseToggle) mouseFlag();
  };

  mouseEnterHandler = () => {
    const { mouseToggle, fenceToggle } = this.props;

    if (fenceToggle && mouseToggle) this.fenceHandler();
  };

  mouseDownHandler = () => {
    const { mouseToggle, mouseFlag, fenceToggle } = this.props;
    if (fenceToggle && !mouseToggle) {
      mouseFlag();
      this.fenceHandler();
    } else {
      this.clickHandler();
    }
  };

  clickHandler = () => {
    const { gridHasStart, gridHasFinish, resetStartFinish } = this.props;
    const { start, finish } = this.state;

    if (gridHasStart && gridHasFinish) {
      // visual aid
      document.getElementById('run-btn').innerText = 'Click Me!';

      let reset = start ? 'start' : finish ? 'finish' : null;

      if (start || finish) resetStartFinish(reset);
    } else if (!gridHasStart) {
      if (finish) alert("Start and end can't be the same");
      else this.startFinishHandler('start');
    } else if (gridHasStart && !gridHasFinish) {
      if (start) {
        alert("Start and end can't be the same");
      } else {
        this.startFinishHandler('finish');
      }
    }
  };

  render() {
    const { start, finish, fence } = this.state;
    const { size } = this.props;
    return (
      <div
        onMouseDown={this.mouseDownHandler}
        onMouseEnter={this.mouseEnterHandler}
        onMouseUp={this.mouseUpHandler}
        style={
          size === 0 ? null : { width: parseInt(size), height: parseInt(size) }
        }
        className={`Node ${
          start ? 'start' : finish ? 'finish' : fence ? 'fence' : ''
        }`}
        id={this.props.id}
      />
    );
  }
}

export { Node };

Node.propTypes = {
  fenceToggle: PropTypes.bool.isRequired,
  gridId: PropTypes.object.isRequired,
  gridHasStart: PropTypes.bool.isRequired,
  gridHasFinish: PropTypes.bool.isRequired,
  mouseFlag: PropTypes.func.isRequired,
  mouseToggle: PropTypes.bool.isRequired,
  nodeFlag: PropTypes.func.isRequired,
  resetStartFinish: PropTypes.func.isRequired,
  size: PropTypes.number.isRequired,
};
