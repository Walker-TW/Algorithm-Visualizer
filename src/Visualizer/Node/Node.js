import React, { Component } from 'react';
import PropTypes from 'prop-types';

// import "./Node.css";
import './NodeTron.css';
// import './AnimatedNode.css';
export default class Node extends Component {
  state = {
    start: false,
    finish: false,
    fence: false,
  };

  // start and end points are not adjustable
  // at the moment so they have a seperate handler

  startFinishHandler = (type) => {
    const { gridId, nodeFlag } = this.props;
    nodeFlag(gridId, [type]);
    this.setState({ [type]: true });
  };

  fenceHandler = () => {
    const { gridId, nodeFlag } = this.props;
    const { fence, start, finish } = this.state;

    if (!start && !finish) nodeFlag(gridId, 'fence');
    this.setState({ fence: !fence });
  };

  mouseUpHandler = () => {
    const { mouseFlag, mouseToggle, fenceToggle } = this.props;

    if (fenceToggle && mouseToggle) mouseFlag();
  };

  mouseEnterHandler = () => {
    const { mouseToggle, fenceToggle } = this.props;

    if (fenceToggle && mouseToggle) this.fenceHandler();
  };

  mouseDownHandler = () => {
    const { mouseToggle, mouseFlag, fenceToggle } = this.props;
    if (fenceToggle && !mouseToggle) {
      mouseFlag();
      this.fenceHandler();
    } else {
      this.clickHandler();
    }
  };

  clickHandler = () => {
    const { gridHasStart, gridHasFinish, resetStartFinish } = this.props;
    const { start, finish } = this.state;

    if (gridHasStart && gridHasFinish) {
      // visual aid
      document.getElementById('run-btn').innerText = 'Click Me!';
      if (start) resetStartFinish('start');
      if (finish) resetStartFinish('finish');
    }
    if (!gridHasStart) {
      if (finish) {
        alert("Start and end can't be the same");
      } else {
        this.startFinishHandler('start');
      }
    }
    if (gridHasStart && !gridHasFinish) {
      if (start) {
        alert("Start and end can't be the same");
      } else {
        this.startFinishHandler('finish');
      }
    }
  };

  render() {
    const { start, finish, fence } = this.state;
    return (
      <div
        onMouseDown={this.mouseDownHandler}
        onMouseEnter={this.mouseEnterHandler}
        onMouseUp={this.mouseUpHandler}
        className={`Node ${
          start ? 'start' : finish ? 'finish' : fence ? 'fence' : ''
        }`}
        id={this.props.id}
      />
    );
  }
}

Node.propTypes = {
  gridId: PropTypes.object.isRequired,
  gridHasStart: PropTypes.bool.isRequired,
  gridHasFinish: PropTypes.bool.isRequired,
  nodeFlag: PropTypes.func.isRequired,
  mouseFlag: PropTypes.func.isRequired,
  mouseToggle: PropTypes.bool.isRequired,
  fenceToggle: PropTypes.bool.isRequired,
  resetStartFinish: PropTypes.func.isRequired,
};
