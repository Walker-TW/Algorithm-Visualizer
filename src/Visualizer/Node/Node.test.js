import React from 'react';
import Node from './Node';
import { shallowToJson } from 'enzyme-to-json';

const nodeFlagSpy = jest.fn(() => {});

const defaultProps = {
  fenceToggle: false,
  gridId: { colIndex: 0, rowIndex: 10 },
  gridHasStart: false,
  gridHasFinish: false,
  mouseFlag: () => {},
  mouseToggle: false,
  nodeFlag: nodeFlagSpy,
  resetStartFinish: () => {},
};

describe('<Node />', () => {
  const node = shallow(<Node {...defaultProps} />);
  it('Test that Node renders', () => {
    expect(shallowToJson(node)).toMatchSnapshot();
  });

  it('starts with required states', () => {
    expect(node.state().start).toEqual(false);
    expect(node.state().finish).toEqual(false);
  });

  it('responds to clicks', () => {
    node.simulate('mousedown', 'mouseup');
    expect(node.state().start).toEqual(true);
  });

  it('calls the nodeFlag method with gridId and a type', () => {
    nodeFlagSpy.mockClear();
    node.simulate('mousedown', 'mouseup');
    expect(nodeFlagSpy.mock.calls.length).toEqual(1);
    expect(nodeFlagSpy.mock.calls[0]).toEqual([
      { colIndex: 0, rowIndex: 10 },
      ['start'],
    ]);
  });
});
