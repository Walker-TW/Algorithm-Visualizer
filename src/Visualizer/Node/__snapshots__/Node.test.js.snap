// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Node /> Test that Node renders 1`] = `
<div
  className="Node "
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "height": NaN,
      "width": NaN,
    }
  }
/>
`;
